messages/php/src/Id/UuidIdGenerator.php

```diff
@@ -4,17 +4,41 @@
 
 namespace Cucumber\Messages\Id;
 
-use Ramsey\Uuid\Uuid;
-
 final class UuidIdGenerator implements IdGenerator
 {
+    /**
+     * Generates a v4 (random based) UUID
+     *
+     * @see https://datatracker.ietf.org/doc/html/rfc4122
+     */
     public function newId(): string
     {
-        // defer to ramsey/uuid if it is present
-        if (class_exists(Uuid::class)) {
-            return Uuid::uuid4()->toString();
-        } else {
-            return (new FallbackUuidGenerator())->newId();
-        }
+        $bytes = random_bytes(16);
+
+        /**
+         * set MSBs of octet 8 to 10xxxxxx. to indicate rfc4122 compliance
+         *
+         * @see https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.1
+         */
+        $bytes[8] = $bytes[8]
+            & "\x3F" // 0 the first 2 bits
+            | "\x80" // raise 1st bit to 1
+        ;
+
+        /**
+         * set MSBs of octet 6 to 0100xxxx. to indicate v4
+         *
+         * @see https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.3
+         */
+        $bytes[6] = $bytes[6]
+            & "\x0F" // 0 the first 4 bits
+            | "\x40" // raise 2nd bit to 1
+        ;
+
+        return preg_replace(
+            '/^(.{8})(.{4})(.{4})(.{4})(.{12})$/',
+            '\\1-\\2-\\3-\\4-\\5',
+            bin2hex($bytes),
```

Comment:
Might be faster to join substrings instead of a regex.

Reply:
I guess I can benchmark 👍  Regex are  often unreasonably fast in PHP

Reply:
Regex 2X faster https://3v4l.org/L6Hqb

Reply:
Er, no significant diff

https://3v4l.org/f82j9

(ignore deleted comment)

Reply:
That's because you've put `$bytes = bin2hex(random_bytes(16));` inside the loop. 

If you move that out, regex is actually faster.

Reply:
Very surprising but looks like the implementation of `preg_replace` can more efficiently build the string then php can.

Reply:
Now I need to google if I can bin2hex as part of the regex!

Reply:
Or maybe stop microoptimising :)

Reply:
It's worth spending some time on it, it gets called quite often.

Reply:
I do think the current performance is reasonable enough though.

Reply:
At some point I'll get some proper benchmarking and see where bottlenecks are (separately)